src/compiler/declarationEmitter.ts

```diff
@@ -396,6 +398,12 @@ namespace ts {
                 }
             }
 
+            function emitTypePredicate(type: TypePredicateNode) {
+                writeEntityName(type.parameterName);

```

Comment:
@rbuckton is right it caused on error being outputted. The parameter of the type predicates is referencing a local variable to the function. 

>That said, a parameter name cannot be a qualified name, so I would just use writeTextOfNode here instead of adding a new writeEntityName.

I can use `writeTextOfNode` then we skip one if statement check.

I broke up `writeEntityName` from `emitEntityName` because I thought it was more performant if it doesn't get evaluated every time we encounter `emitEntityName`, which I guess is quite frequent. So I guess I can keep that change?